api/v2.go

```diff
@@ -79,3 +88,78 @@ type V2Service interface {
 type V2CommonIntegration struct {
 	Data v2CommonIntegrationData `json:"data"`
 }
+
+type V2Pagination struct {
+	Rows      int `json:"rows"`
+	TotalRows int `json:"totalRows"`
+	Urls      struct {
+		NextPage string `json:"nextPage"`
+	} `json:"urls"`
+}
+
+// Pagination is the interface that structs should implement to be able
+// to use inside the client.NextPage() function
+type Pagination interface {
+	PageInfo() *V2Pagination
+	ResetPaging()
+}
+
+// NextPage
+//
+// Use this function to access the next page from an API v2 endpoint, the provided
+// response must implement the Pagination interface and when it is passed, it will
+// be overwritten, if the response doesn't have paging information this function
+// returns false and not error
+//
+// Usage: To iterate over all pages
+//
+// ```go
+// var (
+// 		response = api.MachineDetailsResponse{}
+// 		err      = client.V2.Entities.Search(&response, api.SearchFilter{})
+// )
+//
+// for {
+// 		// Use information from response.Data
+// 		fmt.Printf("Data from page: %d\n", len(response.Data))
+//
+// 		pageOk, err := client.NextPage(&response)
+// 		if err != nil {
+// 			fmt.Printf("Unable to access next page, error '%s'", err.Error())
+// 			break
+// 		}
+//
+// 		if pageOk {
+// 			continue
+// 		}
+// 		break
+// }
+// ```
+func (c *Client) NextPage(p Pagination) (bool, error) {
```

Comment:
I'm not sure about the placement of NextPage method on the Client.
I've been thinking a service could be Pageable and implement the NextPage()
Individual structs in that service could be PaginatedResponses and have functions to expose paging details.